app/controllers/users_controller.rb

```diff
@@ -37,13 +37,19 @@ def update
     end
   end
 
-  def destroy
-    @user = User.find(params[:id])
-    authorize! :destroy, @user
+  def confirm_destroy
+    if @user.valid_password?(params[:password])
+      @user.destroy
+      set_flash_message! :notice, :destroyed
+      redirect_to root_path
+    else
+      set_flash_message! :error, :not_destroyed unless params[:password].blank?
```

Comment:
Why don't we display an error message if the user submits an empty password?

Reply:
No reason anymore, you are right.